src/common/vector_operations/vector_cast.cpp

```diff
@@ -600,38 +631,17 @@ static bool EnumCastSwitch(Vector &source, Vector &result, idx_t count, string *
 		break;
 	}
 	case LogicalTypeId::VARCHAR: {
-		if (source.GetVectorType() == VectorType::CONSTANT_VECTOR) {
-			result.SetVectorType(source.GetVectorType());
-		} else {
-			result.SetVectorType(VectorType::FLAT_VECTOR);
-		}
-		for (idx_t i = 0; i < count; i++) {
-			auto src_val = source.GetValue(i);
-			if (src_val.is_null) {
-				result.SetValue(i, Value());
-				continue;
-			}
-			auto str_vec = EnumType::GetValuesInsertOrder(source.GetType());
-			uint64_t enum_idx;
-			switch (enum_physical_type) {
-			case PhysicalType::UINT8:
-				enum_idx = src_val.value_.utinyint;
-				break;
-			case PhysicalType::UINT16:
-				enum_idx = src_val.value_.usmallint;
-				break;
-			case PhysicalType::UINT32:
-				enum_idx = src_val.value_.uinteger;
-				break;
-			default:
-				throw InternalException("ENUM can only have unsigned integers (except UINT64) as physical types");
-			}
-			result.SetValue(i, Value(str_vec[enum_idx]));
-		}
+		EnumToVarchar(source, result, count, enum_physical_type);
+		break;
+	}
+	default: {
+		// Cast to varchar
+		Vector varchar_cast(LogicalType::VARCHAR, count);
+		EnumToVarchar(source, varchar_cast, count, enum_physical_type);
+		// Try to cast from varchar to whatever we wanted before
+		VectorOperations::TryCast(varchar_cast, result, count, error_message, false);
```

Comment:
The `EnumCastSwitch` should take a strict parameter, so it can pass this onto `VectorOperations::TryCast`, instead of always defaulting to false.